### ISSUE TITLE
Clean up layout.context.admin

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#4613]
**Edit:**
Consensus seems to be to keep the context code in separate files after all. Issue title updated. PR attached to make the code  a bit more legitimate. Added `hook_menu()` `file` entries and removed the temporary `module_load_include()` calls

Original post below:
Just realised a small oops: during developing custom contexts, I put all the new code into a "temporary" `layout.contexts.admin.inc` file and included it with `module_load_include('inc', 'layout', 'layout.context.admin')` (and also did the opposite in the other file too, more oops) because `layout.admin.inc` is huge and hard to work with. That seems to have been merged as is. It is administrative code though, so has all right to be in `layout.admin.inc`. Should we merge it in there or leave it out as a special case?

__
PR by @docwilmot: https://github.com/backdrop/backdrop/pull/3310

**Comment by klonos** [comment]
This requires input from our core committers. So pinging @quicksketch @BWPanda @herbdool and @serundeputy

**Comment by BWPanda** [comment]
If this is the first time an additional 'admin' file like that has been used, I'd suggest putting all the code into the proper 'admin' file as suggested (for this upcoming release), then opening a new issue to discuss what to do in future re. "layout.admin.inc is huge and hard to work with".

**Comment by docwilmot** [comment]
> If this is the first time an additional 'admin' file like that has been used...

We did a separate file for flexible layouts, so I guess a precedent has been set there. I'd opt to keep this a separate file.

**Comment by indigoxela** [comment]
@docwilmot looking at your PR makes me think that the issue description needs an update. :wink:

**Comment by docwilmot** [comment]
Can I get a milestone/review please?

**Comment by indigoxela** [comment]
> Can I get a milestone/review please?

Sure, I cleaned up the labels and added a milestone. I'm not able to do any review today, though, so anyone feel free to do that.

**Comment by indigoxela** [comment]
Code-wise this PR is looking good. :+1: 

But I still have questions regarding handling of deleted entities.

Here's an example in the sandbox: http://3310.backdrop.backdrop.qa.backdropcms.org/

The node required for "Custom Layout Test (custompath)" has been deleted, but the menu item "My custom layout" persists. Clicking on that menu item leads to a completely empty page (no error, nothing in dblog).

I'm not sure if it's possible to catch this (e.g. redirect to a 404 page). Currently Backdrop doesn't give any hint to the admin, what's to do now to fix that page. Should it?
If so, what could be helpful? And what to present to the visitor until it's fixed?

(Sorry if this has been discussed before and I missed it.)

**Comment by docwilmot** [comment]
> But I still have questions regarding handling of deleted entities.

Yes that's concerning. Wasnt there a separate issue? If not then I could post one. As far as the cleanup though, can we RTBC this?

**Comment by docwilmot** [comment]
This issue: https://github.com/backdrop/backdrop-issues/issues/4619

**Comment by indigoxela** [comment]
Many thanks for pointing me to that issue. I forgot that it's still open.

Sure, cleanup and handling of deleted entities are different tasks.

One additional newline in file layout.context.admin.inc and this cleanup PR would be RTBC.

**Comment by indigoxela** [comment]
OK, consensus seems to be that that newline isn't that important. RTBC.

**Comment by quicksketch** [comment]
Thanks folks! I believe at the time we merged the new `layout.context.admin.inc`, I paused a minute on the use of a separate file but I thought it was a good idea because the `layout.admin.inc` is positively huge. This tiny fix-up to make the file only included when necessary is a great improvement and legitimizes the way it's being included.

I've merged https://github.com/backdrop/backdrop/pull/3310 into 1.x and 1.17.x. Thanks!
